tests/EnsController.test.ts

```diff
@@ -0,0 +1,257 @@
+		const controller = new EnsController();
+		expect(controller.set('1', name1, address1)).toBeTruthy();
+		expect(controller.set('1', name1, address1)).toBeFalsy();
+		expect(controller.state).toEqual({
+			ensEntries: {
+				1: {
+					[name1]: {
+						address: address1Checksum,
+						chainId: '1',
+						ensName: name1
+					}
+				}
+			}
+		});
+	});
+
+	it('should not update an ENS entry if the address is the same (null) and return false', () => {
+		const controller = new EnsController();
+		expect(controller.set('1', name1, null)).toBeTruthy();
+		expect(controller.set('1', name1, null)).toBeFalsy();
+		expect(controller.state).toEqual({
+			ensEntries: {
+				1: {
+					[name1]: {
+						address: null,
+						chainId: '1',
+						ensName: name1
+					}
+				}
+			}
+		});
+	});
+
+	it('should add multiple ENS entries and update without side effects', () => {
+		const controller = new EnsController();
+		expect(controller.set('1', name1, address1)).toBeTruthy();
+		expect(controller.set('1', name2, address2)).toBeTruthy();
+		expect(controller.set('2', name1, address1)).toBeTruthy();
+		expect(controller.set('1', name1, address3)).toBeTruthy();
+		expect(controller.state).toEqual({
+			ensEntries: {
+				1: {
+					[name1]: {
+						address: address3Checksum,
+						chainId: '1',
+						ensName: name1
+					},
+					[name2]: {
+						address: address2Checksum,
+						chainId: '1',
+						ensName: name2
+					}
+				},
+				2: {
+					[name1]: {
+						address: address1Checksum,
+						chainId: '2',
+						ensName: name1
+					}
+				}
+			}
+		});
+	});
+
+	it('should get ENS entry by chainId and ensName', () => {
+		const controller = new EnsController();
+		expect(controller.set('1', name1, address1)).toBeTruthy();
+		expect(controller.get('1', name1)).toEqual({
+			address: address1Checksum,
+			chainId: '1',
+			ensName: name1
+		});
+	});
+
+	it('should return null when getting nonexistent name', () => {
+		const controller = new EnsController();
+		expect(controller.set('1', name1, address1)).toBeTruthy();
+		expect(controller.get('1', name2)).toEqual(null);
+	});
+
+	it('should return null when getting nonexistent chainId', () => {
+		const controller = new EnsController();
+		expect(controller.set('1', name1, address1)).toBeTruthy();
+		expect(controller.get('2', name1)).toEqual(null);
+	});
+
+	it('should throw on attempt to set invalid ENS entry: chainId', () => {
+		const controller = new EnsController();
+		expect(() => {
+			controller.set('a', name1, address1);
+		}).toThrowError();
+		expect(controller.state).toEqual({ ensEntries: {} });
+	});
+
+	it('should throw on attempt to set invalid ENS entry: ENS name', () => {
+		const controller = new EnsController();
+		expect(() => {
+			controller.set('1', 'foo.eth', address1);
+		}).toThrowError();
+		expect(controller.state).toEqual({ ensEntries: {} });
+	});
+
+	it('should throw on attempt to set invalid ENS entry: address', () => {
```

Comment:
what is the difference in this case for storing the ens name with a null address?

Reply:
See: https://github.com/MetaMask/gaba/pull/151/files#diff-6dd7c90d258f7dad2ee66f4b66b90c34R107-R118

Reply:
null address indicates that the name did not resolve

Reply:
in this test case it's just an invalid entry :)